pstoremgr/pstoremgr.go

```diff
@@ -243,41 +244,50 @@ func (pm *Manager) LoadPeerstore() (addrs []ma.Multiaddr) {
 
 // SavePeerstore stores a slice of multiaddresses in the peerstore file, one
 // per line.
-func (pm *Manager) SavePeerstore(pinfos []peer.AddrInfo) {
+func (pm *Manager) SavePeerstore(pinfos []peer.AddrInfo) error {
 	if pm.peerstorePath == "" {
-		return
+		err := errors.New("empty peerstore path")
+		logger.Error(err)
+		return err
 	}
 
 	pm.peerstoreLock.Lock()
 	defer pm.peerstoreLock.Unlock()
 
 	f, err := os.Create(pm.peerstorePath)
 	if err != nil {
-		logger.Warningf(
+		logger.Errorf(
 			"could not save peer addresses to %s: %s",
 			pm.peerstorePath,
 			err,
 		)
-		return
+		return err
 	}
 	defer f.Close()
 
 	for _, pinfo := range pinfos {
+		if len(pinfo.Addrs) == 0 {
+			err = errors.New("address info does not have any multiaddresses")
+			logger.Error(err)
+			return err
```

Comment:
don't abort the whole peerstore-writing operation because you found one peer without any multiaddresses, warn about it and move on. This code is used on different places and might be called with things straight from the peerstore which may have no addresses (because maybe they have expired).

You do want to error when reading the list from the flag specifically.